### PR TITLE
Improves Cryoing

### PR DESCRIPTION
**Pull request by Kyep**
Changes:
- While in a cryopod, if you use OOC->Ghost and hit 'Yes', you will be instantly moved to long-term storage. 
- When someone enters long-term storage in a cryopod, the message that announces this now includes their rank. e.g: "John Doe (Captain) has entered long term storage"
- When an antag target cryos, the message that they get about 'Time for plan B' is now more obvious.

Please note:
- You cannot be cryoed against your will. Triggering the instant-storage requires you to deliberately enter a cryopod, choose 'Ghost' from the OOC menu, and then hit 'Yes' on the 'are you sure' prompt. By that point, it is pretty clear you want to leave the round. 
- On the other hand, moving you to long-term storage faster helps other people. It frees the job slot for others, reassigns antag targets, lets everyone know what happened, updates the manifest, etc.

:cl: Kyep
tweak: Entering a cryo pod, then deliberately selecting OOC->Ghost, and clicking yes on the prompt, will now instantly send you to long-term storage, removing you from the round, despawning your body, freeing your job slot, and giving a new objective to any antag that had you as a target.
tweak: Announcements about someone entering long-term storage via a cryopod now include that person's rank.
tweak: When an antag target cryos, the message the antag gets about their objectives changing is now clearer.
/:cl:

### CODE REVIEW

**Comment by monster860** [comment]
Should probably get rid of the useless `list()` inside the `pick()`

**Comment by KasparoVy** [comment]
So if I put an already-SSD guy in cryo, they'll instantly go into storage?

**Comment by Kyep** [comment]
@monster860 Done.

File: code/modules/mob/dead/observer/observer.dm [hunk 1]
```diff
@@ -157,13 +157,17 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
...
 		var/mob/dead/observer/ghost = ghostize(0)            //0 parameter is so we can never re-enter our body, "Charlie, you can never come baaaack~" :3
 		ghost.timeofdeath = world.time // Because the living mob won't have a time of death and we want the respawn timer to work properly.
-
 	var/obj/structure/morgue/Morgue = locate() in M.loc
 	if(istype(M.loc,/obj/structure/morgue))
 		Morgue = M.loc
 	if(Morgue)
 		Morgue.update()
-
+	if(istype(M.loc,/obj/machinery/cryopod))
```
Comment (Fox-McCloud) [inline code/modules/mob/dead/observer/observer.dm]: `if(istype(M.loc, /obj/machinery/cryopod))`

**Comment by Kyep** [comment]
@Fox-McCloud Changed

**Comment by FalseIncarnate** [comment]
I rarely use cryodorms myself, but I can definitely stand behind this tweak. This will be especially useful for people who cryo at roundstart because their job will be reopened MUCH faster, which is particularly helpful for roles like doctors and engineers.

:+1: for also making sure it won't auto-affect SSD people. That would be terribly inconvienent if you crash and a Good Samaritan tosses you into cryo dorms thinking you left.

**Comment by Belario** [comment]
:+1:  For adding actual efficiency.

It's a real pain when half the members of _certain_ departments all run off and there's no room for replacements, not really a problem but this can only result in goods things.

**Pull request merged**